File: ios/Bahandi/Stores/AuthStore.swift
import SwiftUI

enum AuthStatus { case loading, authed, guest }

@MainActor
final class AuthStore: ObservableObject {
    @Published var user: User?
    @Published var status: AuthStatus = .loading

    var role: String { user?.role ?? "" }

    func restoreSession() async {
        guard TokenStore.access != nil else { status = .guest; return }
        do {
            user = try await APIClient.shared.me().user
            status = .authed
        } catch {
            TokenStore.clear()
            status = .guest
        }
    }

    /// Логин. Бросает APIError при неуспехе (для показа ошибки в форме).
    func login(identifier: String, password: String) async throws {
        let res = try await APIClient.shared.login(identifier: identifier, password: password)
        TokenStore.access = res.accessToken
        TokenStore.refresh = res.refreshToken
        user = res.user
        status = .authed
    }

    func logout() {
        TokenStore.clear()
        user = nil
        status = .guest
    }
}

File: front/src/store/authStore.js
import { create } from 'zustand';
import { tokenStore } from '../api/client';
import * as authApi from '../api/auth.api';

// status: 'idle' | 'loading' | 'authed' | 'guest'
export const useAuthStore = create((set, get) => ({
  user: null,
  status: 'idle',
  error: null,

  // Восстановление сессии при старте приложения (по сохранённому токену).
  async init() {
    if (!tokenStore.access) {
      set({ status: 'guest' });
      return;
    }
    set({ status: 'loading' });
    try {
      const { user } = await authApi.me();
      set({ user, status: 'authed' });
    } catch {
      tokenStore.clear();
      set({ user: null, status: 'guest' });
    }
  },

  async login(identifier, password) {
    set({ error: null });
    const data = await authApi.login(identifier, password);
    tokenStore.set({ access: data.access_token, refresh: data.refresh_token });
    set({ user: data.user, status: 'authed' });
    return data.user;
  },

  logout() {
    tokenStore.clear();
    set({ user: null, status: 'guest', error: null });
  },

  setError(error) {
    set({ error });
  },

  get role() {
    return get().user?.role || null;
  },
}));
